dashboard/app/controllers/teacher_dashboard_controller.rb

```diff
@@ -13,8 +18,11 @@ class TeacherDashboardController < ApplicationController
   end
 
   def show
-    @section_summary = @section.selected_section_summarize
     @sections = current_user.sections_instructed.map(&:concise_summarize)
+    if @section.nil?
+      @section = Section.find(@sections.first[:id])
```

Comment:
Have you considered what it would take to make the dashboard not rely on having a selected section?

I think how you have it here is a great approach and might be better, I had just been thinking about the above approach before and wondered if you had considered it.